cmd/tempo-cli/cmd-gen-bloom-index.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/google/uuid"
+
+	"github.com/grafana/tempo/tempodb/backend"
+	"github.com/grafana/tempo/tempodb/encoding"
+	"github.com/grafana/tempo/tempodb/encoding/common"
+)
+
+type genIndexCmd struct {
+	TenantID string `arg:"" help:"tenant-id within the bucket"`
+	BlockID  string `arg:"" help:"block ID to list"`
+	backendOptions
+}
+
+func ReplayBlockAndGetRecords(meta *backend.BlockMeta, filepath string) ([]common.Record, error, error) {
+	v, err := encoding.FromVersion(meta.Version)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var warning error
+	// replay file to extract records
+	f, err := os.OpenFile(filepath, os.O_RDONLY, 0644)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	dataReader, err := v.NewDataReader(backend.NewContextReaderWithAllReader(f), meta.Encoding)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer dataReader.Close()
+
+	var buffer []byte
+	var records []common.Record
+	objectReader := v.NewObjectReaderWriter()
+	currentOffset := uint64(0)
+	for {
+		buffer, pageLen, err := dataReader.NextPage(buffer)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			warning = err
+			break
+		}
+
+		reader := bytes.NewReader(buffer)
+		id, _, err := objectReader.UnmarshalObjectFromReader(reader)
+		if err != nil {
+			warning = err
+			break
+		}
+
+		// make a copy so we don't hold onto the iterator buffer
+		recordID := append([]byte(nil), id...)
+		records = append(records, common.Record{
+			ID:     recordID,
```

Comment:
Need another set of 👀 to think through this.   This is using the first trace ID from the page for the index entry, but I think AppenderBuffered uses the last ID per page:  https://github.com/grafana/tempo/blob/main/tempodb/encoding/appender_buffered.go#L52   Thoughts?

Reply:
In the process of implementing an Iterator for the page. Will update tomorrow.

Reply:
Yes, agree with Marty. The index entry is the largest id on the page.